### PR TITLE
ENH: Add BIDS standard types

### PR DESCRIPTION
**Pull request by larsoner**
Adds two new channel types:

1. galvanic skin response
2. temperature

Quoting @hoechenberger:

> BIDS spec:
> https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/03-electroencephalography.html#channels-description-_channelstsv
>
> Issue was discovered & discussed here:
> https://github.com/mne-tools/mne-bids/issues/1046

### CODE REVIEW

**Comment by mkajola** [comment]
Hello Eric & co,

Haven’t had time to look much, but maybe its time to see what could be pulled in.

Is that 2. Skin temperature? I suppose the question to decide is if a generic temperature is enough, or do we need to specialize it.
The second one may lead to quite large number of channel types, but would have clearer definition.

t. mjk





From: Eric Larson ***@***.***>
Sent: torstai 25. elokuuta 2022 15.33
To: mne-tools/fiff-constants ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [mne-tools/fiff-constants] ENH: Add BIDS standard types (PR #41)


Adds two new channel types:

  1.  galvanic skin response
  2.  temperature

Quoting @hoechenberger<https://github.com/hoechenberger>:

BIDS spec:
https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/03-electroencephalography.html#channels-description-_channelstsv

Issue was discovered & discussed here:
mne-tools/mne-bids#1046<https://github.com/mne-tools/mne-bids/issues/1046>

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/mne-tools/fiff-constants/pull/41

Commit Summary

  *   eea9e29<https://github.com/mne-tools/fiff-constants/pull/41/commits/eea9e291577b022b3a326ebe6124af0e0c2bbe11> ENH: Add BIDS standard types

File Changes

(1 file<https://github.com/mne-tools/fiff-constants/pull/41/files>)

  *   M DictionaryTypes.txt<https://github.com/mne-tools/fiff-constants/pull/41/files#diff-eb9545a3fbc15857df72863b5a9a516e8c305bc20e08eaa12966c89eed434264> (2)

Patch Links:

  *   https://github.com/mne-tools/fiff-constants/pull/41.patch
  *   https://github.com/mne-tools/fiff-constants/pull/41.diff

**Comment by larsoner** [comment]
The "Temperature" is generic in the BIDS spec, so it probably makes sense to make it generic here, too. I'm not sure if we'd need different types for different sensors (environmental, skin, ... ?), I'm inclined to just make the basic type here and let people's channel names specify what temperature is being measured.

File: DictionaryTypes.txt [hunk 1]
```diff
@@ -274,6 +274,8 @@ enum(ch_type) "Type of the channel"
     dipole_wave    1000		     "Dipole time curve"
     goodness_fit   1001		     "Goodness of fit"
     fnirs	   1100		     "Functional near-infrared spectroscopy"
+    temperature	   1200		     "Temperature"
+    galvanic	   1300		     "Galvanic skin response"
```
Comment (hoechenberger) [inline DictionaryTypes.txt]: We should name the channel type eda and the description should be Electrodermal activity. GSR is an outdated term that ideally should also be eliminated from BIDS....

See also Wikipedia on the naming:

https://en.wikipedia.org/wiki/Electrodermal_activity?wprov=sfti1

**Review by hoechenberger — CHANGES_REQUESTED** [review]

**Comment by hoechenberger** [comment]
> I'm inclined to just make the basic type here and let people's channel names specify what temperature is being measured.

+1 for doing that

File: DictionaryTypes.txt (hunk 1 above)
Comment (agramfort) [reply DictionaryTypes.txt]: I would suggest we do what BIDS does here. EDA is exploratory data analysis with my prior...

Comment (hoechenberger) [reply DictionaryTypes.txt]: I feel quite strongly we should use EDA. I haven't read GSR in publications in quite a while 

(But of course my literature selection is biased)

Comment (hoechenberger) [reply DictionaryTypes.txt]: @cbrnr @sappelhoff any thoughts?

Comment (sappelhoff) [reply DictionaryTypes.txt]: I am not so deeply entrenched in this nomenclature, but if the info in the wikipedia article is valid that "EDA" is a term to standardize the diverse previous terms, then I am all for it.

On the BIDS side we can (given the assumption is true that EDA is the new standard) then deprecate GSR and declare that people use EDA for future datasets.

In MNE-BIDS we can map both EDA and GSR (bids) to EDA (mne), and warn if we do a GSR->EDA mapping, that GSR is deprecated in BIDS.

Comment (hoechenberger) [reply DictionaryTypes.txt]: I'll ask some colleagues of mine who do (or did) peripheral measurements of anything "emotion", maybe they have an opinion

Comment (hoechenberger) [reply DictionaryTypes.txt]: tl;dr: Both terms are used interchangeably, but there seems to be a consensus that EDA is to be preferred these days

So it seems that EDA and GSR are used interchangeably, but GSR mostly sticks around for historical reasons. In many publications, you'll find phrases like "... measured electrodermal activity (EDA; also known as galvanic skin response, GSR)". Like, this sort of double-naming appears to be kind of omnipresent.

I checked how software packages for EDA processing are named:
- [LEDALAB](https://www.mathworks.com/matlabcentral/fileexchange/68502-ledalab-open-source-matlab-software-for-analysis-of-skin-conductance-data-viz-eda-gsr)
- [pyEDA](https://github.com/HealthSciTech/pyEDA)
- NeuroKit2 has an [`eda` package](https://neuropsychology.github.io/NeuroKit/functions/eda.html)
- there's a Biopack[ Electrodermal Activity Analysis Module](https://www.biopac.com/product/eda-analysis-software-module/) for "EDA/GSR Analysis"

Sensors often come with both both names too:

- [iMotions EDA – Electrodermal Activity (GSR)]( https://imotions.com/biosensor/gsr-galvanic-skin-response-eda-electrodermal-activity/)
- [BrainProducts GSR Sensor](https://www.brainproducts.com/solutions/sensors/#tab-1224120e9da80b7ba16) (no mention of EDA here though!)

In summary, my favorite would be `eda`, but I could live with `gsr` too – the community will understand both.

I veto against `galvanic`, though, as this is too generic, and I suppose it won't ring a bell for many folks.

Comment (sappelhoff) [reply DictionaryTypes.txt]: +1 for `gsr` then, as it'll make our lives a bit easier (we wouldn't have to discuss deprecating `gsr` in favor of `eda` at BIDS) and it seems to stick.

Comment (hoechenberger) [reply DictionaryTypes.txt]: Let's do `gsr` then, this also avoids @agramfort's brain freeze 😅

**Review by agramfort — APPROVED** [review]

**Review by hoechenberger — APPROVED** [review]

**Comment by hoechenberger** [comment]
LGTM, feel free to merge, anyone!
And thanks @larsoner, and @agramfrort and  @sappelhoff for chiming in too

**Review by sappelhoff — APPROVED** [review]

**Comment by larsoner** [comment]
Thanks for the quick reviews / conversations @mkajola @hoechenberger @sappelhoff @agramfort !
